### PR TITLE
[14.0.x] ISPN-15964 Fix out-of-order query request serialization

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-15964

Backport of https://github.com/infinispan/infinispan/pull/12267

### CODE REVIEW

File: remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/impl/Externalizers.java [hunk 1]
```diff
@@ -27,9 +27,9 @@ public void writeObject(ObjectOutput output, QueryRequest queryRequest) throws I
          output.writeUTF(queryRequest.getQueryString());
          output.writeLong(queryRequest.getStartOffset() != null ? queryRequest.getStartOffset() : -1);
          output.writeInt(queryRequest.getMaxResults() != null ? queryRequest.getMaxResults() : -1);
-         output.writeInt(queryRequest.hitCountAccuracy() != null ? queryRequest.hitCountAccuracy() : -1);
          output.writeObject(queryRequest.getNamedParameters());
          output.writeBoolean(queryRequest.isLocal());
+         output.writeInt(queryRequest.hitCountAccuracy() != null ? queryRequest.hitCountAccuracy() : -1);
```
Comment (pruivo) [inline remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/impl/Externalizers.java:32]: aren't these changes breaking backward compatibility? Protostream can handle out-of-order fields, but externalizers don't.

Comment (pruivo) [reply remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/impl/Externalizers.java:32]: IMO, this change should be in the main branch either. I'm guessing this affects clients using jboss-marshalling. cc @tristantarrant

Comment (fax4ever) [reply remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/impl/Externalizers.java:32]: This part of the change is about the @tristantarrant, I would let him reply. I 'm more interested in the other change (the one to Protobuf clients).

**Comment by fax4ever** [comment]
Thank you

**Pull request merged**